lib/middleware.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+var middleware = require('express').Router();
+
+module.exports = function generateMiddleware(options) {
+
+  options = options || {};
+
+  options['cookie-name'] = options['cookie-name'] || 'hof-cookie-check';
+
+  middleware.get('/cookies-required', function renderCookiesRequired(req, res) {
+    res.render('cookies-required');
+  });
+
+  middleware.get('/check-cookies', function checkCookie(req, res) {
+    if (Object.keys(req.cookies).length && req.cookies[options['cookie-name']]) {
+      res.redirect(req.cookies[options['cookie-name']]);
+    } else {
+      res.redirect('/cookies-required');
```

Comment:
We could avoid the second redirect and render immediately. That way we would only need to make the one endpoint. (Less pollution.) What do you think? (I guess it doesn't matter that much.)